### ISSUE TITLE
Inject node not executing at specified time

### ISSUE THREAD

**Issue by cpiekacz** [node-red/node-red#2833]
All of my inject nodes that are set to run in the morning (e.g. 7am on weekdays or 9am everyday) are not being executed.
What’s puzzling is that when I create a new node during the day to run at specific hour (let’s say in 5 minutes or something), it usually gets executed. And on top of that, one day last week those nodes fired up, even though the day before and the following days they did not. All other nodes work as expected.

### What are the steps to reproduce?

Create the flow with inject node set to run everyday at 9am and log a timestamp to a file.

Example flow with cronplus:

`[{"id":"a3348e2c.a9997","type":"api-current-state","z":"f78679f0.c6c6f8","name":"Anyone home","server":"ba0d3ced.3f2b6","version":1,"outputs":2,"halt_if":"on","halt_if_type":"str","halt_if_compare":"is","override_topic":false,"entity_id":"binary_sensor.anyone_home","state_type":"str","state_location":"payload","override_payload":"msg","entity_location":"data","override_data":"msg","blockInputOverrides":false,"x":360,"y":200,"wires":[["86da21aa.bc5dc"],[]]},{"id":"86da21aa.bc5dc","type":"api-call-service","z":"f78679f0.c6c6f8","name":"Turn on","server":"ba0d3ced.3f2b6","version":1,"debugenabled":false,"service_domain":"switch","service":"turn_on","entityId":"switch.coffee_maker","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":540,"y":200,"wires":[[]]},{"id":"6be55c90.9e09d4","type":"cronplus","z":"f78679f0.c6c6f8","name":"Everyday at 9am","outputField":"payload","timeZone":"Europe/Warsaw","persistDynamic":false,"commandResponseMsgOutput":"output1","outputs":1,"options":[{"name":"Everday at 9am","topic":"timestamp","payloadType":"date","payload":"","expressionType":"cron","expression":"0 0 9 * * *","location":"","offset":"0","solarType":"all","solarEvents":"sunrise,sunset"}],"x":130,"y":200,"wires":[["d7084310.10f1e","a3348e2c.a9997"]]},{"id":"5a0a0f59.75099","type":"inject","z":"f78679f0.c6c6f8","name":"Everyday at 9am","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"00 09 * * *","once":false,"onceDelay":0.1,"topic":"Inject test #1 (inject / everyday at 9am)","payload":"","payloadType":"date","x":130,"y":340,"wires":[["d73d2baa.6c0208"]]},{"id":"7ec14de8.1977b4","type":"file","z":"f78679f0.c6c6f8","name":"","filename":"/share/node-red-debug.log","appendNewline":true,"createDir":false,"overwriteFile":"false","encoding":"none","x":700,"y":400,"wires":[[]]},{"id":"bdf58f65.ce754","type":"change","z":"f78679f0.c6c6f8","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"msg.payload & \" - \" & msg.topic","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":440,"y":340,"wires":[["bc8546a5.3cb1f8","7ec14de8.1977b4"]]},{"id":"bc8546a5.3cb1f8","type":"debug","z":"f78679f0.c6c6f8","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":650,"y":340,"wires":[]},{"id":"d73d2baa.6c0208","type":"moment","z":"f78679f0.c6c6f8","name":"","topic":"","input":"payload","inputType":"msg","inTz":"Europe/Warsaw","adjAmount":0,"adjType":"days","adjDir":"add","format":"","locale":"C","output":"","outputType":"msg","outTz":"Europe/Warsaw","x":420,"y":280,"wires":[["bdf58f65.ce754"]]},{"id":"d7084310.10f1e","type":"change","z":"f78679f0.c6c6f8","name":"","rules":[{"t":"set","p":"topic","pt":"msg","to":"Inject test #2 (cronplus / everyday at 9am)","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":200,"y":280,"wires":[["d73d2baa.6c0208"]]},{"id":"ba0d3ced.3f2b6","type":"server","name":"Home Assistant","addon":true}]
`
### What happens?

Only cronplus is executed. See dump from log file created by example flow below:

```
~ $ cat /share/node-red-debug.log 
2021-01-23T08:00:00.088Z - Inject test #2 (cronplus / everyday at 9am)
```
### What do you expect to happen?

Inject node runs at 9 am every daay.

### Please tell us about your environment:

- [ ] Node-RED version: 1.2.7 (first noticed on 1.2.5)
- [ ] Node.js version:
- [ ] npm version:
- [ ] Platform/OS: Hass.io
- [ ] Browser:

**Comment by colinl** [comment]
I think this is a duplicate of #1839

**Comment by cpiekacz** [comment]
Unfortunately not, mine are not executing at all. I had to switch to cronplus completely.

**Comment by knolleary** [comment]
The cron library we use under the covers appears to no longer be maintained.

We should consider migrating over to the same module used by the cron-plus node - https://github.com/jaclarke/cronosjs

**Comment by Steve-Mcl** [comment]
Closed by #2959
